src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -109,16 +123,24 @@ MultiRouterPlanCreate(Query *query, MultiExecutorType taskExecutorType)
 	{
 		ErrorIfModifyQueryNotSupported(query);
 		task = RouterModifyTask(query);
+		jobQuery = query;
```

Comment:
why aren't you using the original query here? That should allow us to do delete some odd code in the router code (onConflict and such)?